lib/runtime/router.js

```diff
@@ -77,24 +77,25 @@ export default class Router {
    * @private
    */
   handle(req, response) {
-    let request = new Request(req);
-    let routes = this.routes[request.method];
-    for (let i = 0; i < routes.length; i += 1) {
-      request.params = routes[i].match(request.path);
-      if (request.params) {
-        request.route = routes[i];
-        break;
-      }
-    }
-    if (!request.route) {
-      return this.handleError(request, response, new Errors.NotFound('Route not recognized'));
-    }
+    this.middleware.run(req, response, (error) => {
```

Comment:
The raw `req` object there is a Node `IncomingMessage` straight from the http server. Express decorates this object with several additional properties and methods, so any middleware that leverage those Express specific additions would fail. That's what our `Request` proxy does - it mimics the Express interface.